x-pack/filebeat/input/httpjson/input_cursor.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package httpjson
+
+import (
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/httpjson/config"
+)
+
+type cursorInput struct {
+	*input
+}
+
+func cursorConfigure(cfg *common.Config) ([]cursor.Source, cursor.Input, error) {
+	conf := config.Default()
+	if err := cfg.Unpack(&conf); err != nil {
+		return nil, nil, err
+	}
+	return newCursorInput(conf)
+}
+
+func newCursorInput(config config.Config) ([]cursor.Source, cursor.Input, error) {
+	input, err := newInput(config)
+	if err != nil {
+		return nil, nil, err
+	}
+	return nil, &cursorInput{input: input}, nil
```

Comment:
Each URL MUST become a `cursor.Source`. The cursor input manager calls `Run` per source. The input will not be run when nil is returned.

The cursor passed to Run will be associated with the source. If a URL is taken by another http input instance, Run might not be called (yet) for the other source.

Due to `Run` being called per Source, we require the `input` instance to be stateless. The input should only be a 'configured' immutable 'computation' that can be executed multiple times and in parallel. This requires all state to be held for the lifetime of `Run` only.